csi/csi.go

```diff
@@ -43,12 +44,13 @@ type OsdCsiServerConfig struct {
 // proxies CSI requests for a single specific driver
 type OsdCsiServer struct {
 	Server
-	listener net.Listener
-	server   *grpc.Server
-	driver   volume.VolumeDriver
-	wg       sync.WaitGroup
-	running  bool
-	lock     sync.Mutex
+	listener    net.Listener
+	server      *grpc.Server
+	driver      volume.VolumeDriver
+	wg          sync.WaitGroup
+	running     bool
+	lock        sync.Mutex
```

Comment:
Does the lock protect access to all members ? 
If so embed the lock within the struct.
If not comments can explain what is being protected using this lock.

Reply:
locks access to start and stop only